### PR TITLE
Various code optimizations

### PR DESCRIPTION
**Pull request by samdark**
See code review for explanation.

### CODE REVIEW

File: src/Request.php [hunk 1]
```diff
@@ -16,11 +16,11 @@
     use RequestTrait;
 
     /**
-     * @param string                               $method  HTTP method
-     * @param string|UriInterface                  $uri     URI
-     * @param array                                $headers Request headers
-     * @param string|null|resource|StreamInterface $body    Request body
-     * @param string                               $version Protocol version
+     * @param string $method HTTP method
```
Comment (samdark) [inline src/Request.php]: Original comment was not parsable by PhpStorm.

File: src/Uri.php [hunk 2]
```diff
@@ -73,11 +73,11 @@ public function getAuthority(): string
 
         $authority = $this->host;
         if ('' !== $this->userInfo) {
-            $authority = $this->userInfo.'@'.$authority;
+            $authority = $this->userInfo . '@' . $authority;
```
[the remark this answers is not in the document]
Comment (samdark) [reply src/Uri.php]: But it seems your code style differs... is that correct?

File: src/Uri.php [hunk 3]
```diff
@@ -319,10 +309,10 @@ private function filterQueryAndFragment($str): string
             throw new \InvalidArgumentException('Query and fragment must be a string');
         }
 
-        return \preg_replace_callback('/(?:[^'.self::$charUnreserved.self::$charSubDelims.'%:@\/\?]++|%(?![A-Fa-f0-9]{2}))/', [$this, 'rawurlencodeMatchZero'], $str);
+        return \preg_replace_callback('/(?:[^' . self::CHAR_UNRESERVED . self::CHAR_SUB_DELIMS . '%:@\/\?]++|%(?![A-Fa-f0-9]{2}))/', [__CLASS__, 'rawurlencodeMatchZero'], $str);
     }
 
-    private function rawurlencodeMatchZero(array $match): string
+    private static function rawurlencodeMatchZero(array $match): string
```
Comment (samdark) [inline src/Uri.php]: Made it static. Should be a bit faster...

**Comment by Nyholm** [comment]
I'll move ahead and merge this. 
@Zegnat I will not tag this until you given a thumbs up.

**Comment by Nyholm** [comment]
Thank you @samdark

**Pull request merged**